k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -75,6 +94,67 @@ impl UpgradeArgs {
             }
         }
     }
+
+    ///  Dummy upgrade the resources.
+    pub async fn dummy_apply(&self, namespace: &str) -> Result<(), Error> {
```

Comment:
Can the dummy apply be done by the apply itself by passing a flag? This would ensure the dummy apply is following exactly what the apply does?

Reply:
The code for upgrade,  helm upgrade, draining node, restarting io-engine pod etc are done in upgade-job code. This only executes when the job is created. with dry run we are not creating any resource, so the code cant flow to server side to get the dry run o/p. Everything here is done at client side.

Reply:
Is the code not on he same crate?

Reply:
no, the upgrade code is in upgrade-job crate for which @niladrih  has the PR.